src/libraries/System.Collections.Immutable/Directory.Build.props

```diff
@@ -3,5 +3,7 @@
   <PropertyGroup>
     <StrongNameKeyId>Microsoft</StrongNameKeyId>
     <IsNETCoreApp>true</IsNETCoreApp>
+    <!-- Set CONTRACTS_FULL to preserve usage of Contracts attributes -->
+    <DefineConstants>$(DefineConstants),CONTRACTS_FULL</DefineConstants>
```

Comment:
I assume the assembly doesn't use any of the methods like Contract.Requires that require a binary rewriter when this is specified?

Reply:
I see no usage of anything else in System.Diagnostics.Contracts other than PureAttribute.

Reply:
A long time ago, we used those other contract methods, but they were since removed.